src/foam/dao/index/AltIndex.js

```diff
@@ -32,21 +41,79 @@ foam.CLASS({
 
   properties: [
     {
+      /** delegate factories */
       name: 'delegates',
       factory: function() { return []; }
-    }
+    },
+    {
+      /** factory quick lookup */
+      name: 'delegateMap_',
+      factory: function() { return {}; }
+    },
   ],
 
   methods: [
-    function addIndex(index) {
-      // Populate the index
-      var a = foam.dao.ArraySink.create();
-      this.plan(a).execute([Promise.resolve()], a);
 
-      index.bulkLoad(a);
-      this.delegates.push(index);
+    /** Returns smallest estimate from the delegates */
+    function estimate(size, sink, skip, limit, order, predicate) {
+      var cost = Number.MAX_VALUE;
+      for ( var i = 0; i < this.delegates.length; i++ ) {
+        cost = Math.min(
+          cost,
+          this.delegates[i].estimate(
+            size, sink, skip, limit, order, predicate)
+        );
+      }
+      return cost;
+    },
+
+    function toPrettyString(indent) {
+      var ret = "";
+      for ( var i = 0; i < this.delegates.length; i++ ) {
+          ret += this.delegates[i].toPrettyString(indent + 1);
+      }
+      return ret;
+    },
+
+    function toString() {
+      return 'Alt([' + (this.delegates.join(',')) + '])';
+    },
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.dao.index',
+  name: 'AltIndexNode',
```

Comment:
Every index needs nodes. In the simple case you could assemble an index that would have one chain of Indexes, with one single instance of IndexNode for each.

If a TreeIndex appears in the chain, then each TreeNode will store a separate IndexNode for the tail index. So in that case a Tree -> Alt setup would have one TreeIndexNode holding many AltIndexNodes, each of which may or may not have the same set of delegates.

The AltIndex instance that manages all those nodes knows about every possible delegate type, and can estimate which one to use (and build, if missing from a particular delegate node) for selects.